dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -193,10 +210,14 @@ private static ArtifactCache loadArtifactInfo(List<Artifact> artifacts) {
     return cache;
   }
 
-  private static ArtifactResults generateArtifactReport(Configuration configuration, Path output,
-      Artifact artifact, ArtifactInfo artifactInfo, List<DependencyGraph> globalDependencies)
-      throws IOException, TemplateException, DependencyCollectionException,
-      DependencyResolutionException {
+  private static ArtifactResults generateArtifactReport(
+      Configuration configuration,
+      Path output,
+      Artifact artifact,
+      ArtifactInfo artifactInfo,
+      List<DependencyGraph> globalDependencies,
+      ImmutableMap<Path, JarLinkageReport> pathToLinkageReport)
+      throws IOException, TemplateException {
```

Comment:
generateArtifactReport does not throw DependencyCollectionException or DependencyResolutionException. No logic was removed from the function;